### PR TITLE
[runtime] Initialize thread suspend policy at startup, inline getter

### PR DESCRIPTION
**Pull request by lambdageek**
On some benchmarks we saw `mono_threads_suspend_policy()` using a lot of CPU time.

Rather than doing an init-on-first-use, initialize the policy once upfront when the JIT it starting up and make the getter method static inline.

### CODE REVIEW

File: mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -4080,6 +4080,8 @@ mini_init (const char *filename, const char *runtime_version)
 
 	CHECKED_MONO_INIT ();
 
+	mono_threads_suspend_policy_init ();
+
```
Comment (vargaz) [inline mono/mini/mini-runtime.c]: Shouldn't this be called from like mono_thread_info_init () ?

Comment (lambdageek) [reply mono/mini/mini-runtime.c]: Yea makes sense. I'll move it.

File: mono/utils/mono-threads-coop.c [hunk 2]
```diff
@@ -703,7 +702,7 @@ mono_threads_suspend_validate_policy (MonoThreadsSuspendPolicy policy)
 void
 mono_threads_suspend_override_policy (MonoThreadsSuspendPolicy new_policy)
 {
-	threads_suspend_policy = (char)mono_threads_suspend_validate_policy (new_policy);
+	mono_threads_suspend_policy_hidden_dont_modify = (char)mono_threads_suspend_validate_policy (new_policy);
```
Comment (jaykrell) [inline mono/utils/mono-threads-coop.c]: Avoid contractions as too informal? `do_not` instead of `dont`.

Comment (jaykrell) [reply mono/utils/mono-threads-coop.c]: There are other names for this in practise also, like "mostly readonly". abbreviated imho obscurely as "mr".
Often times such data is actually made read only after initialization with VirtualProtect. Though there are considerations for that, i.e. isolating it to a page with also mostly readonly.

Comment (jaykrell) [reply mono/utils/mono-threads-coop.c]: You can also play useful games like declaring it as const but implementating it as non const.
With externC. i.e. to enforce not writing to it, except where allowed.

**Comment by lambdageek** [comment]
@monojenkins build

**Pull request merged**